tests/transforms/test_get_unitary_matrix.py

```diff
@@ -0,0 +1,296 @@
+    expected_state2 = teststate2
+
+    get_matrix = get_unitary_matrix(testcircuit, wires)
+    matrix = get_matrix()
+
+    obtained_state1 = matrix @ teststate1
+    obtained_state2 = matrix @ teststate2
+
+    assert np.allclose(obtained_state1, expected_state1)
+    assert np.allclose(obtained_state2, expected_state2)
+
+
+# Test MultiControlledX
+def test_get_unitary_matrix_MultiControlledX():
+    """Test with many control wires"""
+    wires = [0, 1, 2, 3, 4, 5]
+
+    def testcircuit():
+        qml.MultiControlledX(control_wires=[0, 2, 4, 5], wires=3)
+
+    state0 = [1, 0]
+    state1 = [0, 1]
+
+    teststate1 = reduce(np.kron, [state1, state1, state1, state1, state1, state1])
+    teststate2 = reduce(np.kron, [state0, state1, state0, state0, state1, state0])
+
+    expected_state1 = reduce(np.kron, [state1, state1, state1, state0, state1, state1])
+    expected_state2 = teststate2
+
+    get_matrix = get_unitary_matrix(testcircuit, wires)
+    matrix = get_matrix()
+
+    obtained_state1 = matrix @ teststate1
+    obtained_state2 = matrix @ teststate2
+
+    assert np.allclose(obtained_state1, expected_state1)
+    assert np.allclose(obtained_state2, expected_state2)
+
+
+def test_get_unitary_matrix_input_tape():
+    """Test with quantum tape as input"""
+    with qml.tape.QuantumTape() as tape:
+        qml.RX(0.432, wires=0)
+        qml.RY(0.543, wires=0)
+        qml.CNOT(wires=[0, 1])
+        qml.RX(0.133, wires=1)
+
+    get_matrix = get_unitary_matrix(tape)
+
+    matrix = get_matrix()
+
+    part_expected_matrix = np.kron(qml.RY(0.543, wires=0).matrix @ qml.RX(0.432, wires=0).matrix, I)
+
+    expected_matrix = np.kron(I, qml.RX(0.133, wires=1).matrix) @ CNOT @ part_expected_matrix
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+def test_get_unitary_matrix_input_QNode():
+    """Test with QNode as input"""
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.qnode(dev)
+    def my_quantum_function():
+        qml.PauliZ(wires=0)
+        qml.CNOT(wires=[0, 1])
+        qml.PauliY(wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+    get_matrix = get_unitary_matrix(my_quantum_function)
+
+    matrix = get_matrix()
+
+    expected_matrix = np.kron(I, Y) @ CNOT @ np.kron(Z, I)
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+def test_get_unitary_matrix_input_QNode_wires():
+    """Test with QNode as input, and nonstandard wire ordering"""
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.qnode(dev)
+    def my_quantum_function():
+        qml.PauliZ(wires=0)
+        qml.CNOT(wires=[0, 1])
+        qml.PauliY(wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+    get_matrix = get_unitary_matrix(my_quantum_function, wire_order=[1, 0])
+    matrix = get_matrix()
+
+    CNOT10 = np.array([[1, 0, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0], [0, 1, 0, 0]])
+
+    expected_matrix = np.kron(Y, I) @ CNOT10 @ np.kron(I, Z)
+
+    assert np.allclose(matrix, expected_matrix)
+
```

Comment:
You could maybe add a more complicated test with more operations acting on different qubits (e.g. 4 qubits and 5 different operations), otherwise it is really good 👍

Reply:
Sure! Thanks.